### PR TITLE
Type ``infer_unary_op``

### PR DESCRIPTION
**Pull request by DanielNoord**
## Steps

- [x] Write a good description on what the PR does.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Related Issue

### CODE REVIEW

**Review by Pierre-Sassoulas — APPROVED** [review]

**Comment by cdce8p** [comment]
@DanielNoord Can you resolve the merge conflict here?

File: astroid/nodes/node_classes.py [hunk 1]
```diff
@@ -1865,6 +1867,8 @@ class Const(_base_nodes.NoChildrenNode, Instance):
 
     _other_fields = ("value", "kind")
 
+    infer_unary_op: ClassVar[InferUnaryOp[Const]]
+
```
Comment (cdce8p) [inline astroid/nodes/node_classes.py:1871]: Can you move the type definitions after `postinit`? Similar to the other PR

**Review by cdce8p — CHANGES_REQUESTED** [review]

File: astroid/nodes/node_classes.py [hunk 2]
```diff
@@ -68,14 +69,15 @@ def _is_const(value):
...
-    typing.Generator[InferenceResult, None, Optional[InferenceErrorInfo]],
-]
 InferBinaryOperation = Callable[
     [_NodesT, Optional[InferenceContext]],
     typing.Generator[Union[InferenceResult, _BadOpMessageT], None, None],
 ]
+InferLHS = Callable[
+    [_NodesT, Optional[InferenceContext]],
+    typing.Generator[InferenceResult, None, Optional[InferenceErrorInfo]],
+]
```
[the remark this answers is not in the document]
Comment (DanielNoord) [reply astroid/nodes/node_classes.py:79]: Intended, I thought I would fix the alphabetic order 😄

[the remark this answers is not in the document]
Comment (cdce8p) [reply astroid/nodes/node_classes.py:79]: 👌🏻

**Review by cdce8p — APPROVED** [review]
Looks good! I would like to merge #1666 first though as this would fix the pylint Home Assistant tests.

**Pull request merged**